### ISSUE TITLE
devtools-service trace always timesout after some action is performed on the page

### ISSUE THREAD

**Issue by Dileep17** [webdriverio/webdriverio#4963]
**Environment :**
 - **WebdriverIO version:** 5.18.5  
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync 
 - **Node.js version:** v11.13.0
 - **NPM version:** 6.13.6
 - **Browser name and version:** Chrome 79.0.3945.130
 - **Platform name and version:** Mac
 - **Additional wdio packages used (if applicable):** @wdio/devtools-service:@5.16.16

**Describe the bug**
 I need to get the perf metrics on various page of my site (includes login, navigating through various pages etc). Modified the perf test example [here](https://github.com/christian-bromann/webdriverio-performance-testing/blob/master/tests/performance.test.js) to fit my spike as below.
```
let metrics, score
describe('My Performance Test', () => {
    before(() => {
        browser.enablePerformanceAudits()
    })

    it('should not increase firstInteractive limi', function () {
        browser.url('https://magnificent-cabbage.glitch.me')
        metrics = browser.getMetrics()
        score = browser.getPerformanceScore()
        console.log(metrics.firstInteractive) 
        console.log(metrics.speedIndex)
        console.log(score)
        $('=PROS & CONS').click()
        metrics = browser.getMetrics()
        score = browser.getPerformanceScore()
        console.log(metrics.firstInteractive) 
        console.log(metrics.speedIndex)
        console.log(score)
    })
})
```
Irrespective of the timeout set, this test always fails with below error
`Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

getMetrics and getPerformanceScore worked for the first time (after page is loaded) but trace failed(timed out!) after click on page is performed.  (`Wait until tracing for command click finishes`)

**Log**
```
...
[0-0] 2020-01-17T06:50:07.675Z INFO @wdio/devtools-service: Disable throttling
[0-0] 2020-01-17T06:50:07.677Z INFO @wdio/devtools-service: continuing with next WebDriver command
[0-0] 2020-01-17T06:50:07.678Z INFO webdriver: COMMAND getMetrics()
[0-0] 2020-01-17T06:50:07.960Z INFO webdriver: COMMAND getPerformanceScore()
[0-0] 2020-01-17T06:50:07.961Z INFO webdriver: RESULT { estimatedInputLatency: 16,
  timeToFirstByte: 563,
  domContentLoaded: 3079,
  firstVisualChange: 3181,
  firstPaint: 3275,
  firstContentfulPaint: 3275,
  firstMeaningfulPaint: 3275,
  lastVisualChange: 4781,
  firstCPUIdle: 3275,
  firstInteractive: 3275,
  load: 4762,
  speedIndex: 4118 }
[0-0] 3275
[0-0] 4118
0.8320207881193711
[0-0] 2020-01-17T06:50:07.962Z INFO webdriver: RESULT 0.8320207881193711
[0-0] 2020-01-17T06:50:07.964Z INFO webdriver: COMMAND findElement("link text", "PROS & CONS")
[0-0] 2020-01-17T06:50:07.964Z INFO webdriver: [POST] http://127.0.0.1:4444/session/a47d30931dc24352190358545b0e056f/element
[0-0] 2020-01-17T06:50:07.964Z INFO webdriver: DATA { using: 'link text', value: 'PROS & CONS' }
[0-0] 2020-01-17T06:50:07.978Z INFO webdriver: RESULT { 'element-6066-11e4-a52e-4f735466cecf': 'b5415e14-2f8d-482f-b852-f0b1c8b18ddf' }
[0-0] 2020-01-17T06:50:07.983Z INFO @wdio/devtools-service:TraceGatherer: Start tracing frame with url click transition
[0-0] 2020-01-17T06:50:07.988Z INFO @wdio/devtools-service:TraceGatherer: Waiting for CPU to become idle
[0-0] 2020-01-17T06:50:07.989Z INFO webdriver: COMMAND elementClick("b5415e14-2f8d-482f-b852-f0b1c8b18ddf")
[0-0] 2020-01-17T06:50:07.989Z INFO webdriver: [POST] http://127.0.0.1:4444/session/a47d30931dc24352190358545b0e056f/element/b5415e14-2f8d-482f-b852-f0b1c8b18ddf/click
[0-0] 2020-01-17T06:50:08.051Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 10597.484999998414 ms
[0-0] 2020-01-17T06:50:08.051Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 10597.484999998414 ms
[0-0] 2020-01-17T06:50:08.176Z INFO @wdio/devtools-service: Wait until tracing for command click finishes
[0-0] Error in "My Performance Test should not increase firstInteractive limi"
Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/fe/Documents/Learning/wdio/test/specs/sample.js)
[0-0] 2020-01-17T06:50:24.223Z INFO webdriver: COMMAND deleteSession()
[0-0] 2020-01-17T06:50:24.224Z INFO webdriver: [DELETE] http://127.0.0.1:4444/session/a47d30931dc24352190358545b0e056f
[0-0] FAILED in chrome - /test/specs/sample.js
2020-01-17T06:50:24.399Z INFO @wdio/cli:launcher: Run onComplete hook
...
```

Tried running example provided [here](https://webdriver.io/docs/devtools-service.html#performance-testing)
```
    const assert = require('assert')
    describe('JSON.org page', () => {
         before(() => {
             browser.enablePerformanceAudits()
         })

      it('should load within performance budget', () => {
        browser.url('http://json.org')
        let metrics = browser.getMetrics()
        assert.ok(metrics.speedIndex < 1500) // check that speedIndex is below 1.5ms
        let score = browser.getPerformanceScore() // get Lighthouse Performance score
        assert.ok(score >= .99) // Lighthouse Performance score is at 99% or higher

        $('=Esperanto').click()
        metrics = browser.getMetrics()
        assert.ok(metrics.speedIndex < 1500)
        score = browser.getPerformanceScore()
        assert.ok(score >= .99)
       })

       after(() => {
          browser.disablePerformanceAudits()
      })
    })
```
This test is also failing for the same reason but I get more clear error. Im not sure if above is related.
Log
```
...
2020-01-17T07:20:43.829Z INFO @wdio/devtools-service:TraceGatherer: Waiting on https://www.json.org/Programma-Bold.woff2 to finish
[0-0] 2020-01-17T07:20:44.109Z INFO @wdio/devtools-service:TraceGatherer: Found 1 inflight network records
[0-0] 2020-01-17T07:20:44.109Z INFO @wdio/devtools-service:TraceGatherer: Waiting on https://www.json.org/img/string.png to finish
[0-0] 2020-01-17T07:20:44.124Z INFO @wdio/devtools-service:TraceGatherer: Found 0 inflight network records
[0-0] 2020-01-17T07:20:44.127Z INFO @wdio/devtools-service:TraceGatherer: Found 1 inflight network records
2020-01-17T07:20:44.127Z INFO @wdio/devtools-service:TraceGatherer: Waiting on https://www.json.org/favicon.gif to finish
[0-0] 2020-01-17T07:20:44.696Z INFO @wdio/devtools-service:TraceGatherer: Found 0 inflight network records
[0-0] 2020-01-17T07:20:48.834Z INFO @wdio/devtools-service:TraceGatherer: Network became finally idle
[0-0] 2020-01-17T07:20:48.835Z INFO @wdio/devtools-service:TraceGatherer: Wait for network idle canceled
[0-0] 2020-01-17T07:20:51.321Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 8741.054999998596 ms
[0-0] 2020-01-17T07:20:51.322Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 8741.054999998596 ms
[0-0] 2020-01-17T07:20:52.635Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 10054.859999996552 ms
[0-0] 2020-01-17T07:20:52.635Z INFO @wdio/devtools-service:TraceGatherer: Driver CPU has been idle for 10054.859999996552 ms
[0-0] 2020-01-17T07:20:52.636Z INFO @wdio/devtools-service:TraceGatherer: Tracing completed after 12389ms, capturing performance data for frame 397F6EDF368F85E569CB90E6E83225FD
[0-0] 2020-01-17T07:20:52.637Z INFO @wdio/devtools-service:TraceGatherer: Tracing completed after 12389ms, capturing performance data for frame 397F6EDF368F85E569CB90E6E83225FD
[0-0] (node:22998) UnhandledPromiseRejectionWarning: Error: Protocol error (IO.read): Read failed
    at Promise (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:182:12)
    at Function.readProtocolStream (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/helper.js:254:37)
    at CDPSession._client.once.event (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Tracing.js:64:14)
    at Object.onceWrapper (events.js:281:20)
    at CDPSession.emit (events.js:198:15)
    at CDPSession.EventEmitter.emit (domain.js:481:20)
    at CDPSession._onMessage (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:200:12)
    at Connection._onMessage (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:112:17)
[0-0] (node:22998) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:22998) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[0-0] (node:22998) UnhandledPromiseRejectionWarning: Error: Protocol error (IO.read): Read failed
    at Promise (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:183:56)
    at new Promise (<anonymous>)
    at CDPSession.send (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:182:12)
    at Function.readProtocolStream (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/helper.js:254:37)
    at CDPSession._client.once.event (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Tracing.js:64:14)
    at Object.onceWrapper (events.js:281:20)
    at CDPSession.emit (events.js:198:15)
    at CDPSession.EventEmitter.emit (domain.js:481:20)
    at CDPSession._onMessage (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:200:12)
    at Connection._onMessage (/Users/fe/Documents/Learning/wdio/node_modules/puppeteer-core/lib/Connection.js:112:17)
[0-0] (node:22998) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
[0-0] Error in "JSON.org page should load within performance budget"
Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/fe/Documents/Learning/wdio/test/specs/s1.js)
[0-0] 2020-01-17T07:21:10.249Z INFO webdriver: COMMAND disablePerformanceAudits()
[0-0] 2020-01-17T07:21:10.250Z INFO webdriver: RESULT undefined
[0-0] 2020-01-17T07:21:10.254Z INFO webdriver: COMMAND deleteSession()
[0-0] 2020-01-17T07:21:10.254Z INFO webdriver: [DELETE] http://127.0.0.1:4444/session/eebb1f86da5f9c6ef143082c41e2610d
...
```

It would be really helpful if someone can check this. Thanks in advance.

**Comment by christian-bromann** [comment]
> Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

Have you tried to increase the timeout? Performance tests are usually a bit slow as it takes time to properly fetch all metrics.

**Comment by Dileep17** [comment]
@christian-bromann yeah. increased timeout but no luck.

**Comment by tomaraks** [comment]
Facing same issue.

**Comment by stevenlafl** [comment]
Same problem.

**Comment by christian-bromann** [comment]
This seems indeed like bug. We would appreciate any help on this so please help us out! Have a look at our [contributing guidelines](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) and reach out if you have any questions. Cheers!

**Comment by christian-bromann** [comment]
This bug should be fixed with the latest version of `v7.14.0`. Please report if not. Closing.
